shared/src/main/scala/carldata/series/Csv.scala

```diff
@@ -33,22 +38,61 @@ object Csv {
 
   /** Reader for CSV string */
   def fromString(str: String, dateFormatter: DateTimeFormatter = defaultFormatter): Seq[TimeSeries[Double]] = {
-    val data = str.split("\n").tail.map { line =>
-      val tokens = line.split(",")
-      (LocalDateTime.parse(tokens(0), dateFormatter).toInstant(ZoneOffset.UTC), tokens.tail.map(_.toDouble).toVector)
-    }.toVector
-
-    data.unzip._2
-      .transpose
-      .map(vs => TimeSeries(data.unzip._1, vs))
+    val lines = str.split(newLineDelimeter)
+    require(lines.nonEmpty)
+    val header = lines.head
+    val columnsCount = header.split(tokensDelimeter) match {
+      case Array("time", "value") => 2
+      case arr if arr.size > 2 && arr(0) == "time" &&
+        arr.zipWithIndex.tail.forall { case (p, i) => p == s"value$i" } => arr.size
+      case _ => throw new IllegalArgumentException("Wrong input string format")
+    }
+    val valueColumnsCount = columnsCount - 1
+
+    val data = Vector.fill(valueColumnsCount)(ArrayBuffer.empty[(Long, Double)])
```

Comment:
I think we do not have to use mutable ArrayBuffer if `data` is result of recursive function.